microprofile-open-api/src/main/java/org/jboss/eap/qe/microprofile/openapi/apps/routing/router/rest/LocalServiceRouterInfoResource.java

```diff
@@ -0,0 +1,41 @@
+package org.jboss.eap.qe.microprofile.openapi.apps.routing.router.rest;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.eclipse.microprofile.openapi.annotations.media.Content;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponses;
+
+/**
+ * Rest resource exposing operations which provide information about Local Service Router, this is a "non-routed"
+ * service - i.e. only related to the Local Service Router domain
+ */
+@Path("/info")
+public class LocalServiceRouterInfoResource {
+
+    /**
+     * Returns local service router FQDN
+     *
+     * @return Response containing the local service router FQDN
+     */
+    @GET
+    @Path("/fqdn")
+    @Produces(MediaType.TEXT_PLAIN)
+    @APIResponses(value = {
+            @APIResponse(responseCode = "200", description = "Local Service Provider FQDN", content = @Content(mediaType = MediaType.TEXT_PLAIN)),
+            @APIResponse(responseCode = "500", description = "An error occurred while retrieving Local Service Provider FQDN") })
+    @Operation(summary = "Get local service router FQDN", description = "Retrieves and returns the local service router FQDN", operationId = "getFullyQualifiedDomainName")
+    public Response getFullyQualifiedDomainName() {
+        try {
+            return Response.ok(InetAddress.getLocalHost().getHostName()).build();
+        } catch (UnknownHostException e) {
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
```

Comment:
The `500` error code is not send when an exception is thrown on server so that's why are you returning it?

Reply:
I get what you mean, probably this was definitely not necessary. Fixing, thanks.